### ISSUE TITLE
AnimatedObjectOnce performance concern

### ISSUE THREAD

**Issue by minchemo** [RafaelBarbosatec/bonfire#359]
It always showing these finished `animatedObject`, no matter how I call visibleComponents anywhere.
I am concerning these will lead to a terrible performance once there are lot player in my game, will these auto remove from game or I can ignore these ?

```dart
Iterable<GameComponent> areas = gameRef.visibleComponents();
inspect(areas);
```

![3](https://user-images.githubusercontent.com/22622775/208810351-5cb1a82a-40eb-48db-971c-1e5e0708c638.png)

This is a `stateController` which used to make any animation.

(I have init others animation in SimplePlayer but I have to do something at `onStart` and `onFinish`, so I call `playOnce` instead of `playOther`)
```dart
class LocalPlayerController extends StateController<LocalPlayer> {
/// other code...
  void serveAction(ActionType actionType) {
    component!.animation?.playOnce(
        runToTheEnd: true,
        SpriteAssets().getCreaturesAnimation( // function to get animation
            raceName: component!.race.race!,
            movementName: actionType.toText(),
            forceDirection: component!.lastDirection,
            forceStepTime: 0.1 / component!.race.speed!), 
onStart: () {
// onStart
    }, onFinish: () {
// onFinish
    });
  }

}
```

**Comment by RafaelBarbosatec** [comment]
Hi @minchemo !
It's strange, the method `animation?.playOnce` create AnimatedObjectOnce but not add in the game.
Those AnimatedObjectOnce must be adding in the game for another way.

**Comment by RafaelBarbosatec** [comment]
Fixed in version `2.11.10`
